source/s3_meta_request.c

```diff
@@ -236,9 +236,13 @@ static void s_s3_request_on_finish(
 
     if (request_binding->input_body) {
         /* close the input file stream now, request has finished, we will not read from there anymore */
-        aws_input_stream_destroy(request_binding->input_body);
+        aws_input_stream_release(request_binding->input_body);
```

Comment:
we don't need the `s3_meta_request_binding.input_body` variable at all anymore

delete it

when we create the input_body, we add it to the request. Then we can just release it, the request will keep it alive